rust/storage/src/lib.rs

```diff
@@ -237,6 +237,25 @@ impl Storage {
         }
     }
 
+    pub async fn strongly_consistent_get_with_e_tag(
+        &self,
+        key: &str,
+        options: GetOptions,
+    ) -> Result<(Arc<Vec<u8>>, Option<ETag>), StorageError> {
+        match self {
+            Storage::ObjectStore(object_store) => {
+                object_store.strongly_consistent_get_with_e_tag(key).await
+            }
+            Storage::S3(s3) => s3.get_with_e_tag(key).await,
+            Storage::Local(local) => local.strongly_consistent_get_with_e_tag(key).await,
```

Comment:
nit: for local also you could just call `get` since it will be strongly consistent?

Reply:
Not sure about object store